### PR TITLE
Created carousel component for testimonials

### PR DESCRIPTION
**Pull request by subhajit20**
Created carousel component for testimonials
Issue - #2370

### CODE REVIEW

**Comment by subhajit20** [comment]
@dtslvr Hey I have created the carousel component.
Just need help with how to place that carousel component on the home page.

**Comment by dtslvr** [comment]
> Just need help with how to place that carousel component on the home page.

Can you integrate an example below the current testimonials [here](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/landing/landing-page.html#L321)?

File: libs/ui/src/lib/carousel/carousel.ts [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    ViewEncapsulation,
+  } from '@angular/core';
+  import {FocusableOption, FocusKeyManager} from '@angular/cdk/a11y';
+  import {LEFT_ARROW, RIGHT_ARROW, TAB} from '@angular/cdk/keycodes';
+  import {ANIMATION_MODULE_TYPE} from '@angular/platform-browser/animations';
+  import {MatIconModule} from '@angular/material/icon';
+  import {MatButtonModule} from '@angular/material/button';
+  import {NgIf} from '@angular/common';
+  
+  @Directive({
```
Comment (dtslvr) [inline libs/ui/src/lib/carousel/carousel.ts:22]: Can you extract this part to new file `carousel-item.directive.ts`?

File: libs/ui/src/lib/carousel/carousel.ts [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+  
+    constructor(readonly element: ElementRef<HTMLElement>) {}
+  
+    focus(): void {
+      this.element.nativeElement.focus({preventScroll: true});
+    }
+  }
+  
+  @Component({
+    selector: 'app-carousel',
```
Comment (dtslvr) [inline libs/ui/src/lib/carousel/carousel.ts:38]: Can you use our prefix: `gf-carousel`

**Review by dtslvr — CHANGES_REQUESTED** [review]

File: libs/ui/src/lib/carousel/carousel.ts (hunk 1 above)
Comment (dtslvr) [reply libs/ui/src/lib/carousel/carousel.ts:22]: > only I have to copy that 4 lines and paste that into `carousel-item.directive.ts`?

No, from line 22 to line 35

Comment (subhajit20) [reply libs/ui/src/lib/carousel/carousel.ts]: done this.
please check and let me know.

Comment (dtslvr) [reply libs/ui/src/lib/carousel/carousel.ts]: Thanks 👍🏻

**Review by dtslvr — CHANGES_REQUESTED** [review]
Can you integrate an example below the current testimonials [here](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/landing/landing-page.html#L321)?

File: libs/ui/src/lib/carousel/carousel.scss [hunk 3]
```diff
@@ -0,0 +1,37 @@
+app-carousel {
```
Comment (dtslvr) [inline libs/ui/src/lib/carousel/carousel.scss:1]: This needs to be changed to `gf-carousel` as well...

**Comment by subhajit20** [comment]
> Can you integrate an example below the current testimonials [here](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/landing/landing-page.html#L321)?

Can you help out with how I can place that carousel component?

**Comment by dtslvr** [comment]
Add something like this to the landing page:

```
<gf-carousel>
  <div *ngFor="let testimonial of testimonials" carousel-item>
    {{ testimonial.quote }}
  </div>
</gf-carousel>
```

Original source from material.angular.io: [homepage.html](https://github.com/angular/material.angular.io/blob/main/src/app/pages/homepage/homepage.html#L54)

Have you been able to run Ghostfolio on your local machine?

**Comment by subhajit20** [comment]
> > Can you help out with how I can place that carousel component?
> 
> Add something like this to the landing page:
> 
> ```
> <gf-carousel>
>   <div *ngFor="let testimonial of testimonials" carousel-item>
>     {{ testimonial.quote }}
>   </div>
> </gf-carousel>
> ```
> 
> Original source from material.angular.io: [homepage.html](https://github.com/angular/material.angular.io/blob/main/src/app/pages/homepage/homepage.html#L54)
> 
> Have you been able to run Ghostfolio on your local machine?

Added this code snippet,
Can you check and let me know for further correction?

File: apps/client/src/app/pages/landing/landing-page.html [hunk 4]
```diff
@@ -345,7 +345,25 @@ <h2 class="h4 mb-1 text-center" i18n>
           </div>
         </div>
       </div>
-    </div>
+    </div> -->
+    <gf-carousel [aria-label]="'Featured components'">
+      <a *ngFor="let comp of getTopComponents()" carousel-item>
```
Comment (dtslvr) [inline apps/client/src/app/pages/landing/landing-page.html:350]: Change to:

```
<div *ngFor="let testimonial of testimonials" carousel-item>
```

File: apps/client/src/app/pages/landing/landing-page.html [hunk 5]
```diff
@@ -345,7 +345,25 @@ <h2 class="h4 mb-1 text-center" i18n>
...
+        <gf-logo
+            class="mr-3 mt-2 pt-1"
+            size="medium"
+            [showLabel]="false"
+        ></gf-logo>
+        {{(comp[0].toUpperCase() + comp.slice(1)).replace('-', ' ')}}
```
Comment (dtslvr) [inline apps/client/src/app/pages/landing/landing-page.html:356]: Remove line

File: apps/client/src/app/pages/landing/landing-page.html [hunk 6]
```diff
@@ -345,7 +345,25 @@ <h2 class="h4 mb-1 text-center" i18n>
...
+        <div class="mt-2 text-muted">
+          —
+          <a *ngIf="testimonial.url" target="_blank" [href]="testimonial.url"
+            >{{ testimonial.author }}</a
+          >
+          <span *ngIf="!testimonial.url">{{ testimonial.author }}</span>, {{
+          testimonial.country }}
+        </div>
+      </a>
```
Comment (dtslvr) [inline apps/client/src/app/pages/landing/landing-page.html:365]: Change to:
```
</div>
```

**Review by dtslvr — CHANGES_REQUESTED** [review]
Can you add this to the following file `apps/client/src/app/pages/landing/landing-page.module.ts`:

```
import { CarouselItem } from '@ghostfolio/ui/carousel/carousel-item.directive';
import { Carousel } from '@ghostfolio/ui/carousel/carousel';
```

and change this
```
declarations: [LandingPageComponent],
```
to this:
```
declarations: [Carousel, CarouselItem, LandingPageComponent],
```

File: libs/ui/src/lib/carousel/carousel-item.directive.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { FocusableOption } from "@angular/cdk/a11y";
+import { Directive, ElementRef, HostBinding } from "@angular/core";
+
+@Directive({
+    selector: '[carousel-item]',
+    standalone: true,
```
Comment (dtslvr) [inline libs/ui/src/lib/carousel/carousel-item.directive.ts:6]: Remove line

File: libs/ui/src/lib/carousel/carousel.ts [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+  import {MatButtonModule} from '@angular/material/button';
+  import {NgIf} from '@angular/common';
+import { CarouselItem } from './carousel-item.directive';
+  
+  @Component({
+    selector: 'gf-carousel',
+    templateUrl: './carousel.html',
+    styleUrls: ['./carousel.scss'],
+    encapsulation: ViewEncapsulation.None,
+    standalone: true,
```
Comment (dtslvr) [inline libs/ui/src/lib/carousel/carousel.ts:28]: Remove line

**Review by dtslvr — APPROVED** [review]

**Comment by dtslvr** [comment]
Thank you for your changes @subhajit20. I have made some adaptions, now it's ready for merge.

**Comment by subhajit20** [comment]
> Thank you for your changes @subhajit20. I have made some adaptions, now it's ready for merge.

Thanks for your guidance 🙏🏻.
Really appreciated 😁

**Comment by subhajit20** [comment]
hey @dtslvr can you add `hacktoberfest-accepted` label to this pr?

**Pull request merged**